torch/distributed/_sharded_tensor/api.py

```diff
@@ -214,6 +223,76 @@ def _init_rpc(self):
         # Barrier for all RPCs to finish on all ranks.
         rpc.api._barrier(rpc_workers)
 
+    @classmethod
+    def init_from_local_shards(
+        cls,
+        local_shards: List[torch.Tensor],
+        shards_metadata: List[ShardMetadata],
```

Comment:
@pritamdamania I am thinking of whether we should let user pass in `EnumerableShardingSpec` here instead of `List[ShardMetadata]` as they are essentially the same thing. Let me know which one you prefer.

Reply:
Yeah I think adding a `ShardedTensorMetadata` make a lot of sense to me, and grouping the attributes together makes it more intuitive to the user as well I think. This might makes it looks similar but not entirely same concept with the ShardingSpec, but I think we can document it so that user won't get confused.